app/components/AddRuleModal.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from "react";
+import { Modal } from "./ModalWrapper";
+export const AddRuleModal: React.SFC<{}> = props => {
+  return (
+    <Modal show={props.showModal} handleClose={props.handleClose}>
+      <label htmlFor="modal-url">URL</label>
+      <input
+        className="form-control"
+        type="text"
+        name="modal-url"
+        id="url-input-modal"
+        onChange={e => props.updateModalUrl(e.target.value)}
```

Comment:
`onChange={props.updateModalUrl}`.
`updateModalUrl` will receive `event` object as first argument and do the `e.target.value` inside  `updateModalUrl`.

Advantage: avoiding function declaration on reach render.

Reply:
Had missed this point^^. Thanks for reminding. Better to use this pattern everytime we pass event/event properties